### PR TITLE
[TOOLS-2173] Linux arm64 support.

### PR DESCRIPTION
**Pull request by dwelch-spike**

### CODE REVIEW

File: src/main/common.c [hunk 1]
```diff
@@ -575,7 +531,22 @@ key_append(const as_val* key, const as_val* val, void* udata)
 LOCAL_HELPER int
 key_cmp(const void* v1, const void* v2)
 {
-	return as_val_cmp(*(as_val**)v1, *(as_val**)v2);
+	// why offset by 1 here because as_val_cmp considers 0 as less than, 1 as equal, and 2 as greater.
```
Comment (jdogmcsteezy) [inline src/main/common.c:534]: "why offset by 1 . . ." Maybe "We offset by 1"

File: src/main/transaction.c [hunk 2]
```diff
@@ -1126,7 +1128,12 @@ queue_pop_wait(queue_t* adata_q)
 	while (1) {
 		adata = queue_pop(adata_q);
 		if (adata == NULL) {
+			#ifdef __aarch64__
```
Comment (jdogmcsteezy) [inline src/main/transaction.c:1131]: Not sure what is going on here.  Why is this being added?

**Review by jdogmcsteezy — COMMENTED** [review]
Couple questions.

File: src/main/common.c (hunk 1 above)
Comment (dwelch-spike) [reply src/main/common.c]: changed to "we ..."

File: src/main/transaction.c (hunk 2 above)
Comment (dwelch-spike) [reply src/main/transaction.c:1131]: This is the arm64 counterpart to what `_mm_pause()` does. [_mm_pause()](https://android.googlesource.com/platform/prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.7/+/7f61ceb88132c4329380c0cf9634e305dda9fe76/lib/gcc/x86_64-linux-android/4.7/include/xmmintrin.h#1226) decodes to `__asm__ __volatile__ ("rep; nop" : : );` which is the pause instruction on x86. From what I could find it is used to signal to the compiler that we are waiting on a spin lock and some optimizations are possible. I couldn't find a similar function for arm64 so I used the `yield` instruction, which seems to do the same thing. https://developer.arm.com/documentation/dui0802/b/A32-and-T32-Instructions/SEV--SEVL--WFE--WFI--and-YIELD

**Review by jdogmcsteezy — APPROVED** [review]

**Pull request merged**